nmostesting/GenericTest.py

```diff
@@ -126,6 +126,7 @@ def execute_tests(self, test_names):
         """Perform tests defined within this class"""
 
         for test_name in test_names:
+            NMOSUtils.RANDOM.seed(CONFIG.RANDOM_SEED ^ hash(test_name))
```

Comment:
In order that you can run `all` tests and then repeat a single one of the tests,  would it be better to move this into the other overload of `execute_tests` next to the actual `method` calls?

Reply:
you mean in the `execute_test` method in `GenericTest.py` thusly:
```
        # Run manually defined tests
        if test_name == "all":
            for method_name in dir(self):
                if method_name.startswith("test_"):
                    method = getattr(self, method_name)
                    if callable(method):
                        NMOSUtils.RANDOM.seed(CONFIG.RANDOM_SEED ^ hash(method_name))
                        print(" * Running " + method_name)
                        test = Test(inspect.getdoc(method), method_name)
                        try:
                            self.result.append(method(test))
                        except NMOSTestException as e:
                            self.result.append(e.args[0])
                        except Exception as e:
                            self.result.append(self.uncaught_exception(method_name, e))
        # Run a single test
        if test_name != "auto" and test_name != "all":
            method = getattr(self, test_name)
            if callable(method):
                NMOSUtils.RANDOM.seed(CONFIG.RANDOM_SEED ^ hash(method_name))
                print(" * Running " + test_name)
                test = Test(inspect.getdoc(method), test_name)
                try:
                    self.result.append(method(test))
                except NMOSTestException as e:
                    self.result.append(e.args[0])
                except Exception as e:
                    self.result.append(self.uncaught_exception(test_name, e))

Reply:
👍  yes, but also needed in `do_test_api_resource` I think?